Test failing on newer OS X xcode

We expect no stdout/stderr from node-gyp in the tests. Currently therefore they fail since many of the apps print: `warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9` with latest XCode.


To fix this we'll need to apply one of the solutions from the troubleshooting section at https://github.com/mapbox/cpp#os-x-1.

An example failure looks like:

```
 26) complex builds app4 app4 builds with custom --target :

      Uncaught AssertionError: 'clang: warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9\nclang: warning: libstdc+ == ''
      + expected - actual

      -clang: warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9
      -clang: warning: libstdc++ is deprecated; move to libc++ with a minimum deployment target of OS X 10.9
      
      at test/build.test.js:499:32
      at test/build.test.js:40:16
      at ChildProcess.exithandler (child_process.js:204:7)
      at maybeClose (internal/child_process.js:854:16)
      at Process.ChildProcess._handle.onexit (internal/child_process.js:222:5)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,11 +1,13 @@
 language: cpp
 
+sudo: false
+
+# https://docs.travis-ci.com/user/languages/objective-c/#Supported-OS-X-iOS-SDK-versions
+osx_image: xcode8.2
+
 os:
   - linux
-    sudo: false
   - osx
-    # https://docs.travis-ci.com/user/languages/objective-c/#Supported-OS-X-iOS-SDK-versions
-    osx_image: xcode8.2
 
 addons:
   apt:

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,9 +2,10 @@ language: cpp
 
 os:
   - linux
+    sudo: false
   - osx
-
-sudo: false
+    # https://docs.travis-ci.com/user/languages/objective-c/#Supported-OS-X-iOS-SDK-versions
+    osx_image: xcode8.2
 
 addons:
   apt:

--- a/test/app1/binding.gyp
+++ b/test/app1/binding.gyp
@@ -3,17 +3,10 @@
     {
       "target_name": "<(module_name)",
       "sources": [ "<(module_name).cc" ],
-    },
-    {
-      "target_name": "action_after_build",
-      "type": "none",
-      "dependencies": [ "<(module_name)" ],
-      "copies": [
-        {
-          "files": [ "<(PRODUCT_DIR)/<(module_name).node" ],
-          "destination": "<(module_path)"
-        }
-      ]
-    }
+      'product_dir': '<(module_path)',
+      "xcode_settings": {
+        "MACOSX_DEPLOYMENT_TARGET":"10.9"
+      }
+     }
   ]
 }

--- a/test/app1/binding.gyp
+++ b/test/app1/binding.gyp
@@ -5,7 +5,8 @@
       "sources": [ "<(module_name).cc" ],
       'product_dir': '<(module_path)',
       "xcode_settings": {
-        "MACOSX_DEPLOYMENT_TARGET":"10.9"
+        "MACOSX_DEPLOYMENT_TARGET":"10.9",
+        "CLANG_CXX_LIBRARY": "libc++"
       }
      }
   ]

--- a/test/app2/binding.gyp
+++ b/test/app2/binding.gyp
@@ -5,18 +5,11 @@
       "sources": [ "<(module_name).cc" ],
       'include_dirs': [
           '<(custom_include_path)'
-      ]      
-    },
-    {
-      "target_name": "action_after_build",
-      "type": "none",
-      "dependencies": [ "<(module_name)" ],
-      "copies": [
-        {
-          "files": [ "<(PRODUCT_DIR)/<(module_name).node" ],
-          "destination": "<(module_path)/"
-        }
-      ]
+      ],
+      'product_dir': '<(module_path)',
+      "xcode_settings": {
+        "MACOSX_DEPLOYMENT_TARGET":"10.9"
+      }
     }
   ]
 }

--- a/test/app2/binding.gyp
+++ b/test/app2/binding.gyp
@@ -8,7 +8,8 @@
       ],
       'product_dir': '<(module_path)',
       "xcode_settings": {
-        "MACOSX_DEPLOYMENT_TARGET":"10.9"
+        "MACOSX_DEPLOYMENT_TARGET":"10.9",
+        "CLANG_CXX_LIBRARY": "libc++"
       }
     }
   ]

--- a/test/app3/binding.gyp
+++ b/test/app3/binding.gyp
@@ -5,18 +5,11 @@
       "sources": [ "<(module_name).cc" ],
       "dependencies": [
         "deps/mylib.gyp:mylib"
-      ]
-    },
-    {
-      "target_name": "action_after_build",
-      "type": "none",
-      "dependencies": [ "<(module_name)" ],
-      "copies": [
-        {
-          "files": [ "<(PRODUCT_DIR)/<(module_name).node" ],
-          "destination": "<(module_path)/"
-        }
-      ]
+      ],
+      'product_dir': '<(module_path)',
+      "xcode_settings": {
+        "MACOSX_DEPLOYMENT_TARGET":"10.9"
+      }
     }
   ]
 }

--- a/test/app3/binding.gyp
+++ b/test/app3/binding.gyp
@@ -8,7 +8,8 @@
       ],
       'product_dir': '<(module_path)',
       "xcode_settings": {
-        "MACOSX_DEPLOYMENT_TARGET":"10.9"
+        "MACOSX_DEPLOYMENT_TARGET":"10.9",
+        "CLANG_CXX_LIBRARY": "libc++"
       }
     }
   ]

--- a/test/app3/deps/mylib.gyp
+++ b/test/app3/deps/mylib.gyp
@@ -12,6 +12,10 @@
             ],
             'direct_dependent_settings': {
               'include_dirs': [ 'include/' ],
+            },
+            "xcode_settings": {
+                "MACOSX_DEPLOYMENT_TARGET":"10.9",
+                "CLANG_CXX_LIBRARY": "libc++"
             }
         }
     ]

--- a/test/app4/binding.gyp
+++ b/test/app4/binding.gyp
@@ -7,7 +7,8 @@
         "deps/mylib.gyp:mylib"
       ],
       "xcode_settings": {
-        "MACOSX_DEPLOYMENT_TARGET":"10.9"
+        "MACOSX_DEPLOYMENT_TARGET":"10.9",
+        "CLANG_CXX_LIBRARY": "libc++"
       }
     },
     {

--- a/test/app4/binding.gyp
+++ b/test/app4/binding.gyp
@@ -5,7 +5,10 @@
       "sources": [ "<(module_name).cpp" ],
       "dependencies": [
         "deps/mylib.gyp:mylib"
-      ]
+      ],
+      "xcode_settings": {
+        "MACOSX_DEPLOYMENT_TARGET":"10.9"
+      }
     },
     {
       "target_name": "action_after_build",

--- a/test/app4/deps/mylib.gyp
+++ b/test/app4/deps/mylib.gyp
@@ -14,7 +14,9 @@
               'include_dirs': [ 'include/' ],
             },
             'xcode_settings': {
-              'DYLIB_INSTALL_NAME_BASE': '@loader_path/lib.target/'
+              'DYLIB_INSTALL_NAME_BASE': '@loader_path/lib.target/',
+              "MACOSX_DEPLOYMENT_TARGET":"10.9",
+              "CLANG_CXX_LIBRARY": "libc++"
             }
         }
     ]

--- a/test/app5/package.json
+++ b/test/app5/package.json
@@ -10,7 +10,7 @@
     "version": "0.1.0",
     "main": "./index.js",
     "dependencies": {
-      "node-addon-example": "0.1.3",
+      "node-addon-example": "0.1.5",
       "node-pre-gyp": "0.5.x"
     },
     "scripts": {

